src/components/Post.js

```diff
@@ -259,14 +261,64 @@ export class CommentSection extends React.Component {
   static contextTypes = {
     community: object,
     currentUser: object,
-    isProjectRequest: bool
+    isProjectRequest: bool,
+    dispatch: func,
+    socket: object
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      peopleTyping: {}
+    }
+  }
+
+  componentDidMount () {
+    const { post: { id }, expanded } = this.props
+    const { dispatch, socket } = this.context
+    if (expanded) {
+      socket.post(`${upstreamHost}/noo/post/${id}/subscribe`)
+      socket.on('commentAdded', c => dispatch(appendComment(id, c)))
+      socket.on('userTyping', this.userTyping)
+    }
+  }
+
+  componentWillUnmount () {
+    const { post: { id }, expanded } = this.props
+    const { socket } = this.context
+    if (expanded) {
+      socket.post(`${upstreamHost}/noo/post/${id}/unsubscribe`)
+      socket.off('commentAdded')
+      socket.off('userTyping')
+    }
+  }
+
+  userTyping = data => {
+    let newState = this.state
+    if (data.isTyping) {
+      newState.peopleTyping[data.userId] = data.userName
+    } else {
+      delete newState.peopleTyping[data.userId]
+    }
+    this.setState(newState)
+  }
+
+  startedTyping = () => {
```

Comment:
note this syntax which creates a bound function, so we can just refer to `this.startedTyping` elsewhere instead of `this.startedTyping.bind(this)`